api/v1alpha2/source_types.go

```diff
@@ -0,0 +1,104 @@
+// downloaded, and the location to unpack it.
+type BundleDeplopymentSource struct {
+	// Image is the bundle image that backs the content of this bundle.
+	Image *ImageSource `json:"image,omitempty"`
+
+	// Git is the git repository that backs the content of this Bundle.
+	Git *GitSource `json:"git,omitempty"`
+
+	//  HTTP is the remote location that backs the content of this Bundle.
+	HTTP *HTTPSource `json:"http,omitempty"`
+
+	// Relative location to place the fetched artifacts
+	// +optional
+	Destination string `json:"destination,omitempty"`
+}
+
+type ImageSource struct {
+	// Ref contains the reference to a container image containing Bundle contents.
+	Ref string `json:"ref"`
+
+	// Auth configures the authentication method if necessary.
+	Auth *Authentication `json:"auth,omitempty"`
+}
+
+type GitSource struct {
+	// Repository is a URL link to the git repository containing the bundle.
+	// Repository is required and the URL should be parsable by a standard git tool.
+	Repository string `json:"repository"`
+
+	// Directory refers to the location of the bundle within the git repository.
+	// Directory is optional and if not set defaults to `manifests`.
+	//
+	// +kubebuilder:default:=manifests
+	Directory string `json:"directory,omitempty"`
+
+	// Ref configures the git source to clone a specific branch, tag, or commit
+	// from the specified repo. Ref is required, and exactly one field within Ref
+	// is required. Setting more than one field or zero fields will result in an
+	// error.
+	Ref GitRef `json:"ref"`
+
+	// Auth configures the authentication method if necessary.
+	Auth *Authentication `json:"auth,omitempty"`
+}
+
+type GitRef struct {
+	// Branch refers to the branch to checkout from the repository.
+	// The Branch should contain the bundle manifests in the specified directory.
+	Branch string `json:"branch,omitempty"`
+
+	// Tag refers to the tag to checkout from the repository.
+	// The Tag should contain the bundle manifests in the specified directory.
+	Tag string `json:"tag,omitempty"`
+
+	// Commit refers to the commit to checkout from the repository.
+	// The Commit should contain the bundle manifests in the specified directory.
+	Commit string `json:"commit,omitempty"`
+}
+
+type Authentication struct {
+	// Secret contains reference to the secret that has authorization information and is in the namespace that the provisioner is deployed.
+	// The secret is expected to contain `data.username` and `data.password` for the username and password, respectively for http(s) scheme.
+	// Refer to https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret
+	// For the ssh authorization of the GitSource, the secret is expected to contain `data.ssh-privatekey` and `data.ssh-knownhosts` for the ssh privatekey and the host entry in the known_hosts file respectively.
+	// Refer to https://kubernetes.io/docs/concepts/configuration/secret/#ssh-authentication-secrets
```

Comment:
It may be good to add documentation for the authentication against a container registry,  using a kubernetes.io/dockerconfigjson secret?

Reply:
Rukpak seems to currently support only `kubernetes.io/basic-auth` and `kubernetes.io/ssh-auth` (https://github.com/operator-framework/rukpak/blob/main/docs/sources/git.md#private-git-repositories). Support for any other built in [types](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types) doesn't exist yet